Updated Solidity documentation's "building from source" section

bobsummerwill opened the pull request:
It now references install_deps.sh/.bat, and also adds instructions for Windows.

Fixes https://github.com/ethereum/solidity/issues/938

bobsummerwill committed:
Updated Solidity documentation's "building from source" section.
It now references install_deps.sh/.bat, and also adds instructions for Windows.
Cherry picked some further tweaks from Denton-L.  Thanks :-)

Denton-L commented on docs/installing-solidity.rst:
```diff
@@ -32,102 +32,117 @@ Details about the usage of the Node.js package can be found in the
-Ubuntu Trusty (14.04)
----------------------
+Pre-requisites - macOS
+--------------------------------------------------------------------------------
```
Can you make this line the same length as the one above?

Denton-L commented on docs/installing-solidity.rst:
```diff
@@ -32,102 +32,117 @@ Details about the usage of the Node.js package can be found in the
-    sudo apt-get -y install build-essential git cmake libgmp-dev libboost-all-dev \
-        libjsoncpp-dev
+Pre-requisites - Windows
+--------------------------------------------------------------------------------
```
Same with this line. (Do this for all lines?)

VoR0220 commented:
once Denton's requests are completed, I think this is good to merge.

Denton-L commented:
I figured it would be faster to just clean it up a little by myself. Feel free to cherry-pick this in: https://github.com/Denton-L/solidity/tree/building

bobsummerwill commented:
Like that you mean, @Denton-L?

Denton-L commented:
Yeah, looks good!

bobsummerwill commented:
OK - you had some more.   I'll do those too.   2 mins.

bobsummerwill commented:
OK - done now.   Thanks for the tweaks.

I went with square braces for username, because the angled ones do something weird with the markup.

bobsummerwill commented:
OK - Will merge when you give thumbs-up, @Denton-L.

Thanks, @VoR0220 :-)

Denton-L commented on docs/installing-solidity.rst:
```diff
@@ -32,102 +32,119 @@ Details about the usage of the Node.js package can be found in the
+
+Alternatively, you can build for Windows on the command-line, like so:
 
 .. code-block:: bash
```
Sorry, one more little one! 😉

bobsummerwill commented:
Just .. code-block:: bash
should be .. code:: bash, you mean?   On it.

Denton-L commented:
Looks good! Thanks! 👍

chriseth commented on docs/installing-solidity.rst:
```diff
@@ -32,102 +32,119 @@ Details about the usage of the Node.js package can be found in the
-Binary packages of Solidity together with its IDE Mix are available through
-the `C++ bundle <https://github.com/ethereum/webthree-umbrella/releases>`_ of
-Ethereum.
+Binary packages of Solidity available at
```
`are available`?

chriseth replied:
Please also mention at least the ubuntu ppa

chriseth commented on docs/installing-solidity.rst:
```diff
@@ -32,102 +32,119 @@ Details about the usage of the Node.js package can be found in the
 Binary Packages
```
The comment about using browser-solidity without internet is outdated (you cannot just save the page anymore). We can change it to:
`go to https://github.com/ethereum/browser-solidity/tree/gh-pages and download the .zip file as explained on that page.`

chriseth commented on docs/installing-solidity.rst:
```diff
@@ -32,102 +32,119 @@ Details about the usage of the Node.js package can be found in the
+
+.. code:: bash
+
+    cmake --build . --config RelWithDebInfo
```
We could add a last paragraph explaining where the binary will appear and perhaps even telling users to do `sudo make install`.

bobsummerwill committed:
Tweaks as suggested by chriseth.
I will commit this as-is, because it has been pending for a few days over minor issues.
We will continue to iterate on this.

bobsummerwill commented:
Made those suggested updates, @chriseth.
I think the PPA and Homebrew details are correct ...
One the Homebrew side, I think it looks like we should actually create a new `homebrew-solidity` repo to divide things up nicely.   The fact that `geth` and `eth` shared always struck me as a little odd.   Ditto on the PPAs, to be honest.
